tests/roles/pcp_cleanup/tasks/main.yaml

```diff
@@ -2,29 +2,49 @@
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
+    # Cleanup OpenStackControlPlane object
+    oc delete --ignore-not-found=true osctlplane -l core.openstack.org/openstackcontrolplane -n openstack
 
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack-nova-compute-ffu
-    oc delete --ignore-not-found=true --wait=false openstackcontrolplane/openstack
-    oc patch openstackcontrolplane openstack --type=merge --patch '
-    metadata:
-      finalizers: []
-    ' || true
-
-    while oc get pod | grep rabbitmq-server-0; do
-        sleep 2
-    done
-    while oc get pod | grep openstack-galera-0; do
-        sleep 2
+    # Ensure that all pods in openstack namespace are deleted
+    while oc get pod -n openstack | grep -E 'rabbitmq-server-0|openstack-galera-0'; do
```

Comment:
checking for galera is probably ok, there might a situation where the deployment  is gone, but the finalizer on the CR got not removed, which the above you not catch. 

I think checking that there are no openstack related CRs, like the output of the following should not have a result:

```
for i in $(oc get crd -o custom-columns=Name:metadata.name| grep -E "openstack|rabbit"); do oc get $i -o name; done
```

Reply:
I implemented crd/cr removal in the job. So, I removed crd/cr removal from this job